packages/filesystem/src/browser/compressible-tree/compressible-tree.ts

```diff
@@ -0,0 +1,40 @@
+/********************************************************************************
+ * Copyright (c) 2021 SAP SE or an SAP affiliate company and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable } from '@theia/core/shared/inversify';
+import { TreeNode, CompressibleTreeNode, TreeImpl } from '@theia/core/lib/browser';
+
+@injectable()
+export abstract class CompressibleTree extends TreeImpl {
+
+    protected abstract isCompressionEnabled: () => boolean;
```

Comment:
Why do we require a callback instead of just:

```ts
protected abstract isCompressionEnabled(): boolean;
```

Reply:
No strong opinion about having the callback instead of a field, just wanted to know as to why we went for this approach, no need to update :)

Reply:
I had no particular intention, prefered to update...